libindy/src/commands/metrics.rs

```diff
@@ -35,26 +619,53 @@ impl MetricsCommandExecutor {
             }
         };
     }
+
+
     fn _collect(&self) -> IndyResult<String> {
         trace!("_collect >>>");
-        let metrics_map: HashMap<&str, usize> = {
-            let tp_instance = crate::commands::THREADPOOL.lock().unwrap();
-            [
-                (THREADPOOL_ACTIVE_COUNT, tp_instance.active_count()),
-                (THREADPOOL_QUEUED_COUNT, tp_instance.queued_count()),
-                (THREADPOOL_MAX_COUNT, tp_instance.max_count()),
-                (THREADPOOL_PANIC_COUNT, tp_instance.panic_count()),
-                (OPENED_WALLETS_COUNT, self.wallet_service.get_wallets_count()),
-                (OPENED_WALLET_IDS_COUNT, self.wallet_service.get_wallet_ids_count()),
-                (PENDING_FOR_IMPORT_WALLETS_COUNT, self.wallet_service.get_pending_for_import_count()),
-                (PENDING_FOR_OPEN_WALLETS_COUNT, self.wallet_service.get_pending_for_open_count())
-            ].iter().cloned().collect()
-        };
+        let mut metrics_map= serde_json::Map::new();
+        self._append_threapool_metrics(&mut metrics_map);
+        self._append_wallet_metrics(&mut metrics_map);
+        self._append_command_metrics(&mut metrics_map);
         let res = serde_json::to_string(&metrics_map)
-            .to_indy(IndyErrorKind::InvalidState, "Can't serialize a metrics map")?;
+            .to_indy(IndyErrorKind::InvalidState,"Can't serialize a metrics map")?;
 
         trace!("_collect <<< res: {:?}", res);
         debug!("collecting metrics from command thread");
         Ok(res)
     }
+    fn _append_threapool_metrics(&self, metrics_map: &mut Map<String, Value>) {
```

Comment:
I don't know code conventions of libindy, does it require padding private methods with `_`? FYI @jovfer